packages/azure/src/application/AzureAccount.ts

```diff
@@ -116,40 +111,55 @@ export default class AzureAccount extends CloudProviderAccount {
     return R.flatten(estimationResults)
   }
 
-  public async getSubscriptions(): Promise<Subscription[]> {
-    const subscriptions = []
-    for await (const subscription of this.subscriptionClient.subscriptions.list()) {
-      subscriptions.push(subscription)
-    }
-
-    if (subscriptions.length === 0) {
-      this.logger.warn(
-        'No subscription returned for these Azure credentials, be sure the registered application has ' +
-          'enough permissions. Go to https://www.cloudcarbonfootprint.org/docs/azure/ for more information.',
-      )
-    }
-
-    return subscriptions
-  }
-
-  private async getSubscriptionsByIds(
-    subscriptionIds: string[],
+  private async getSubscriptions(
+    subscriptionIds: string[] = [],
   ): Promise<Subscription[]> {
-    const subscriptions = []
+    const AZURE = configLoader().AZURE
+    const defaultAzureSubscriptionIds = subscriptionIds.length
+      ? subscriptionIds
+      : AZURE.SUBSCRIPTIONS
 
-    for (const subscriptionId of subscriptionIds) {
-      try {
-        const subscription = await this.subscriptionClient.subscriptions.get(
-          subscriptionId,
-        )
+    const getSubscriptions = async (): Promise<Subscription[]> => {
```

Comment:
I put these 2 functions as local functions inside getSubscriptions because I don't think their should be multiple methods to get a subscription, only 1 and then determine which one to get based on the param and env variable